packages/core/src/routes/session.ts

```diff
@@ -28,60 +29,108 @@ import { AnonymousRouter } from './types';
 
 export default function sessionRoutes<T extends AnonymousRouter>(router: T, provider: Provider) {
   router.post(
-    '/session',
+    '/session/sign-in/username-password',
+    koaGuard({ body: object({ username: string(), password: string() }) }),
+    async (ctx, next) => {
+      const { username, password } = ctx.guard.body;
+      await signInWithUsernameAndPassword(ctx, provider, username, password);
```

Comment:
nit: as @simeng-li suggested, we can now extract those code here. We can create and issue.

Reply:
attached is the [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1644980418553059).
should we do this at the time we refactor register and sign-in logic?